### PR TITLE
Remove `linear` parameter from `lax.cond_p`

### PR DESCRIPTION
**Pull request by dfm**
As far as I can tell, it seems like the `linear` parameter in the `lax.cond_p` primitive only exists for historical reasons. It could be used for type checking in `_cond_transpose`, but that was removed because of #14026. With this in mind, we could stop tracking this parameter as implemented in this PR, unless we expect that we'd want to re-introduce the type checking in the future.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Thanks, DFM!

File: jax/_src/lax/control_flow/conditionals.py [hunk 1]
```diff
@@ -732,9 +709,9 @@ def transposed(*args):
 
   return _make_closed_jaxpr(transposed, res_avals + jaxpr.out_avals)
 
-def _cond_transpose(cts, *args, branches, linear):
-  del linear  # could use for error checking, but see #14026
+def _cond_transpose(cts, *args, branches):
   index, *ops = args
+  assert type(index) is not ad.UndefinedPrimal
```
Comment (dfm) [inline jax/_src/lax/control_flow/conditionals.py:714]: This is an unrelated drive-by fix. This will never trigger during autodiff, but it can occur when using `linear_transpose` incorrectly (`cond` is not linear in the predicate!). This will at least help identify where the issue is coming from if anyone ever tries that again...
